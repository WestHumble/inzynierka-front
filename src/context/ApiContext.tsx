import React, {createContext, useState} from 'react';
import axios from 'axios';
import {useNavigation} from "@react-navigation/native";
export const ApiContext = createContext(null);

const API_ADDRESS = process.env.EXPO_PUBLIC_API_URL;
export const ApiProvider = ({children}) => {
    const [userToken, setUserToken] = useState<string>();
    const [userRefreshToken, setUserRefreshToken] = useState<string>();
    let axiosAuthInstance, axiosNoAuthInstance

    const deleteInstances = ()=>{
        axiosAuthInstance = null
        axiosNoAuthInstance = null
    }
    async function getInstance(noAuth, successCallback, errorCallback) {
        if (noAuth) {
            if (!axiosNoAuthInstance) {
                axiosNoAuthInstance = axios.create({
                    baseURL: `${API_ADDRESS}`,
                })
            }
            return axiosNoAuthInstance
        }
        if (!axiosAuthInstance) {
            axiosAuthInstance = axios.create({
                baseURL: `${API_ADDRESS}`,
                headers: {
                    Authorization: `Bearer ${userToken}`,
                }
            })
            axiosAuthInstance.interceptors.response.use(
                response => {
                    return response;
                },
                error => {
                    if (error.request && error.request.status !== 401) {
                        throw error;
                    }
                    let failedRequest = error.config;
                    axios.create({
                        baseURL: `${API_ADDRESS}`,
                    }).post('api/token/refresh', {
                        refresh_token: userRefreshToken
                    }).then((res)=>{
                        setUserToken(res.data.token)
                        setUserRefreshToken(res.data.refresh_token)
                        axiosAuthInstance.defaults.headers['Authorization'] = 'Bearer ' + res.data.token;
                        failedRequest.headers['Authorization'] = 'Bearer ' + res.data.token;

                        return axios.request(failedRequest)
                            .then(res => {
                                if (!res || null === successCallback){
                                    return
                                }
                                successCallback(res)
                            })
                            .catch(res => {
                                if (!res || null === errorCallback){
                                    return
                                }
                                errorCallback(res)
                            })
                    }).catch((res)=>{
                        console.error("Unable to refresh token")
                        if (!res || null === errorCallback){
                            return
                        }
                        errorCallback(res)
                    })
                },
            );
        }
        return axiosAuthInstance
    }
    const post = async (endpoint, body? = null, success = null, error = null, noAuth = false)=> {
        if (!noAuth && !userToken){
            return
        }
        (await getInstance(noAuth, success, error)).post(`/api/${endpoint}`, body).then(res => {
            if (!res || null === success){
                return
            }
            success(res)
        }).catch(res => {
            if (!res || null === error){
                return
            }
            error(res)
        })
    }
    const get = async (endpoint, params? = null, success = null, error = null, noAuth = false)=> {
        if (!noAuth && !userToken){
            return
        }
        (await getInstance(noAuth, success, error)).get(`/api/${endpoint}`, {
            params: params
        }).then(res => {
            if (!res || null === success){
                return
            }
            success(res)
        }).catch(res => {
            if (!res || null === error){
                return
            }
            error(res)
        })
    }
    return (
        <ApiContext.Provider value={{post, get, deleteInstances, userToken, setUserToken, setUserRefreshToken}}>
            {children}
        </ApiContext.Provider>
    );
}