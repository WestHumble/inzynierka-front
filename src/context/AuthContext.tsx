import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';
import React, {createContext, useEffect, useState} from 'react';
import { login as loginPost } from "../components/Api/login";


export const AuthContext = createContext(null);

export const AuthProvider = ({children}) => {
    const [isLoading, setIsLoading] = useState(false);
    const [userToken, setUserToken] = useState<string>();

    const login = (email: string, password: string) => {
        setIsLoading(true)
        loginPost(email, password).then(res => {
            setUserToken(res.data.token)
            AsyncStorage.setItem('userToken', res.data.token) 
            setIsLoading(false)
        }).catch(res => {
            console.error(res) 
            setIsLoading(false)
        })
        // axios.post(`http://localhost:81/api/login`, {
        //     email,
        //     password
        // }).then(res => {
        //     setUserToken(res.data.token)
        //     AsyncStorage.setItem('userToken', res.data.token) 
        //     setIsLoading(false)
        // }).catch(res => {
        //     console.error(res) 
        //     setIsLoading(false)
        // })
    }

    const logout = () => {
        setIsLoading(true)
        setUserToken(null)
        AsyncStorage.removeItem('userToken')
        setIsLoading(false)
    }

    const isLoggedIn = async() => {
        try {
            setIsLoading(true)
            let userToken = await AsyncStorage.getItem('userToken')
            setUserToken(userToken == 'null' ? null : userToken)
            setIsLoading(false)
        } catch (e){
            console.log(`isLoggedIn error ${e}`)
        }

    }

    useEffect(() => {
        isLoggedIn()
    }, [])

    return (
        <AuthContext.Provider value={{login, logout, isLoading, userToken}}>
            {children}
        </AuthContext.Provider>
    );
}