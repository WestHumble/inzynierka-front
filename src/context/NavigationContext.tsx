import React, { createContext } from 'react';

export const NavigationContext = createContext(null);
export const NavigationProvider = ({children}) => {
    const navigationRef = React.createRef();

    return (
        <NavigationContext.Provider value={{navigationRef}}>
            {children}
        </NavigationContext.Provider>
    );
}