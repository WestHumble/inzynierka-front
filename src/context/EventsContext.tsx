import React, {createContext, useContext, useEffect, useState} from 'react';
import * as Location from "expo-location";
import {ApiContext} from "./ApiContext";
import {LocationContext} from "./LocationContext";
import {MarkerData} from "../types/marker";
import {Event} from "../types/event";

export const EventsContext = createContext(null);

export const EventsProvider = ({children}) => {
    const [events, setEvents] = useState<Event[]>([]);

    const { userLocation, shareLocation, setShareLocation } =
        useContext(LocationContext);
    const { post, userToken } = useContext(ApiContext);
    const loadCloseEvents = (region) => {
        post('event/close-list', {
            latitude: region?.latitude??userLocation?.coords.latitude ?? 52.4064,
            longitude: region?.longitude??userLocation?.coords.longitude ?? 16.9252,
            distanceInMeters: 1000
        }, (res) => setEvents(res.data))
    };

    useEffect(() => {
        if (userToken) {
            loadCloseEvents()
        } else {
            setEvents([])
        }
    }, [userToken]);
    return (
        <EventsContext.Provider value={{events, loadCloseEvents}}>
            {children}
        </EventsContext.Provider>
    );
}