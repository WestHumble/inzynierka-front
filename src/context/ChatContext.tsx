import React, {createContext, useContext, useEffect, useRef, useState} from 'react';
import {ApiContext} from "./ApiContext";
import * as Notifications from "expo-notifications";
import {NavigationContext} from "./NavigationContext";
import {Chat} from "../types/chat";


export const ChatContext = createContext(null);

export const ChatProvider = ({children}) => {
    const [chatId, setChatId] = useState<number>(null);
    const [initializingChat, setInitializingChat] = useState<boolean>(false);
    const [chats, setChats] = useState<Chat[]>([]);
    const [unreadChatCounter, setUnreadChatCounter] = useState<number>(0);
    const [refreshTime, setRefreshTime] = useState(0);
    const [redirectChatId, setRedirectChatId] = useState<number>(0);
    const { navigationRef } = useContext(NavigationContext);

    const { get, post, userToken } = useContext(ApiContext);
    const initChat = (friends) => {
        setInitializingChat(true)
        post('chat', {
            participantIds: friends.map(f=>f.id)
        }, (res) => {
            getChatList()
            setRedirectChatId(res.data['chatId'])
        })
    };

    const getChatList = () => {
        get('chat/get-all-chats', null, (res) => {
            let chatsCopy = []
            res.data?.forEach((data)=> {
                let chatTmp = chats.find(e => e.id === data.id)

                if (chatTmp) {
                    chatTmp.name = data.name;
                    chatTmp.participants = data.participants;
                    chatTmp.is_seen = data.is_seen;
                    chatTmp.event_id = data.event_id;
                    chatTmp.last_message_date = data.last_message_date;
                    chatsCopy.push(chatTmp)
                } else {
                    chatsCopy.push(data)
                }
            })

            chatsCopy.sort((c1, c2) => c1.last_message_date > c2.last_message_date ? -1 : 1)
            setChats(chatsCopy)
        })
    };

    useEffect(() => {
        if (redirectChatId === null) {
            setInitializingChat(false)
            return
        }
        let chatTmp = chats.find(e => e.id === redirectChatId)

        if (chatTmp) {
            setRedirectChatId(null)
            navigationRef.current?.navigate('Chat', {
                chatId: redirectChatId
            });
        }
    }, [chats, redirectChatId]);
    const setChatMessages = (chatIdParam) => {
        let chatIdUpdate = chatIdParam ?? chatId
        if (chatIdUpdate !== null && undefined !== getChatById(chatIdUpdate)) {
            get('chat/get-all-messages/' + chatIdUpdate, null, (res) => {
                getChatById(chatIdUpdate).messages = res.data?.sort((c1, c2) => c1.created_at > c2.created_at)
                setRefreshTime(Date.now())
            })
        }
    };

    const setChatAsRead = (chatId) => {
        if (chatId !== null) {
            get('chat/seen/' + chatId, null, (res) => {
                getChatList()
            })
        }
    };

    const getChatById = (chatId): Chat => {
        return chats.find(e => e.id === chatId)
    };

    const notificationListener = useRef();
    const responseListener = useRef();

    useEffect(() => {
        notificationListener.current = Notifications.addNotificationReceivedListener(notification => {
            if ('chat_init' === notification.request.trigger.channelId || 'chat' === notification.request.trigger.channelId) {
                if (navigationRef.current?.getCurrentRoute().name === 'Chat' && JSON.parse(
                    JSON.parse(
                        notification.request.trigger.remoteMessage.data.body
                    ).payload
                ).conversation_id === chatId) {
                    setChatAsRead(chatId)
                } else {
                    setTimeout(getChatList, 1000)
                }
            }
        });

        return () => {
            Notifications.removeNotificationSubscription(notificationListener.current);
        };
    }, [chatId, userToken]);

    useEffect(() => {
        responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {
            if ('chat' === response.notification.request.trigger.channelId) {
                navigationRef.current?.navigate('Chat', {
                    chatId: JSON.parse(
                        JSON.parse(
                            response.notification.request.trigger.remoteMessage.data.body
                        ).payload
                    ).conversation_id
                });
            }
        });

        return () => {
            Notifications.removeNotificationSubscription(responseListener.current);
        };
    }, []);

    useEffect(() => {
        if (userToken) {
            getChatList()
        } else {
            setChats([])
        }
    }, [userToken]);

    useEffect(() => {
        setUnreadChatCounter(chats.filter(c=> !c.is_seen).length)
    }, [chats]);

    return (
        <ChatContext.Provider value={{
            chatId,
            setChatId,
            chats,
            getChatList,
            getChatById,
            setChatMessages,
            setChatAsRead,
            unreadChatCounter,
            initChat,
            initializingChat,
            refreshTime,
        }}>
            {children}
        </ChatContext.Provider>
    );
}