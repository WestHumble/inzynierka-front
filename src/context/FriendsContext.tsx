import React, {createContext, useContext, useEffect, useRef, useState} from 'react';
import {ApiContext} from "./ApiContext";
import {AppState} from "react-native";
import {Friend} from "../types/friend";
import * as Notifications from "expo-notifications";
import {FriendInvite} from "../types/friendInvite";
import {NavigationContext} from "./NavigationContext";

export const FriendsContext = createContext(null);

export const FriendsProvider = ({children}) => {
    const [friends, setFriends] = useState<Friend[]>([]);
    const [pendingInvites, setPendingInvites] = useState<FriendInvite[]>([]);
    const [unreadFriendInvitesCounter, setUnreadFriendInvitesCounter] = useState<number>(0);

    const { get, userToken } = useContext(ApiContext);
    const { navigationRef } = useContext(NavigationContext);
    const getFriendList = () => {
        get('user/get-friends', null, (res) => {
            setFriends(res.data)
        })
    };
    const getPendingFriendInvites = () => {
        get('friend-invite/invites', null, (res) => {
            setPendingInvites(res.data)
        })
    };
    const notificationListener = useRef();
    const responseListener = useRef();

    useEffect(() => {
        notificationListener.current = Notifications.addNotificationReceivedListener(notification => {
            if ('new_friend' === notification.request.trigger.channelId) {
                setTimeout(getFriendList, 1000)
            }
            if ('new_friend_invite' === notification.request.trigger.channelId) {
                setTimeout(getPendingFriendInvites, 1000)
            }
        });

        responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {
            if ('new_friend_invite' === response.notification.request.trigger.channelId) {
                navigationRef.current?.navigate('FriendInvitesScreen');
            }
        });

        return () => {
            Notifications.removeNotificationSubscription(notificationListener.current);
            Notifications.removeNotificationSubscription(responseListener.current);
        };
    }, []);

    useEffect(() => {
        if (userToken) {
            getFriendList()
            getPendingFriendInvites()
        } else {
            setFriends([])
            setPendingInvites([])
        }
    }, [userToken]);

    useEffect(() => {
        setUnreadFriendInvitesCounter(pendingInvites.filter(c=> c.status === 'new').length)
    }, [pendingInvites]);

    return (
        <FriendsContext.Provider value={{
            friends,
            getFriendList,
            pendingInvites,
            getPendingFriendInvites,
            unreadFriendInvitesCounter,
        }}>
            {children}
        </FriendsContext.Provider>
    );
}