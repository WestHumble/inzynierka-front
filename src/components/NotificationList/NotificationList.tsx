import React, {useContext} from "react";
import {View, Text, SectionList, StyleSheet, Image} from "react-native";
import CustomButton from "../CustomButton";
import {ApiContext} from "../../context/ApiContext";
import {FriendInvite} from "../../types/friendInvite";
import {FriendsContext} from "../../context/FriendsContext";
import {FriendInviteNotification} from "../../types/notification";
import {EventsContext} from "../../context/EventsContext";
import {EventInvite} from "../../types/eventInvite";
import Avatar from "../Avatar";

const API_ADDRESS = process.env.EXPO_PUBLIC_API_URL;
const FriendInviteItem = ({notification }) => {
    const {post} = useContext(ApiContext);
    const { getPendingFriendInvites, getFriendList } = useContext(FriendsContext);
    const acceptInvite = (friendInvite: FriendInvite) => {
        post('friend-invite/update-status', {
            "friendInviteId": friendInvite.id,
            "transition": "accept"
        }, ()=>{
            getPendingFriendInvites()
            getFriendList()
        })
    }
    const declineInvite = (friendInvite: FriendInvite) => {
        post('friend-invite/update-status', {
            "friendInviteId": friendInvite.id,
            "transition": "decline"
        }, ()=>{
            getPendingFriendInvites()
        })
    }
    return (
        <View style={styles.notificationContainer}>
            <Avatar
                style={{
                    width: 50,
                    height: 50,
                    borderRadius: 5,
                    marginRight: 10,
                    padding: 1,
                    shadowColor: "black",
                    backgroundColor: "#222",
                }}
                userId={notification.friendInvite.friend.id}
                userName={notification.friendInvite.friend.name}
                userSurname={notification.friendInvite.friend.surname}
            />
            <Text
                style={styles.name}>{notification.friendInvite.friend.name} {notification.friendInvite.friend.surname}</Text>
            <CustomButton
                text="Zaakceptuj"
                onPress={() => acceptInvite(notification.friendInvite)}
                type="PRIMARY"
                bgColor={undefined}
                fgColor={undefined}
            />
            <CustomButton
                text="Odrzuć"
                onPress={() => declineInvite(notification.friendInvite)}
                type="PRIMARY"
                bgColor={undefined}
                fgColor={undefined}
            />
        </View>
    )};

const EventInviteItem = ({notification}) => {
    const {post} = useContext(ApiContext);
    const { getPendingEventInvites, loadAllEvents, clearSearchEvents } = useContext(EventsContext);
    const acceptInvite = (eventInvite: EventInvite) => {
        post('event-invite/update-status', {
            "eventInviteId": eventInvite.id,
            "transition": "accept"
        }, ()=>{
            getPendingEventInvites()
            loadAllEvents()
            clearSearchEvents()
        })
    }
    const declineInvite = (eventInvite: EventInvite) => {
        post('event-invite/update-status', {
            "eventInviteId": eventInvite.id,
            "transition": "decline"
        }, ()=>{
            getPendingEventInvites()
        })
    }
    return (
        <View style={styles.notificationContainer}>
            <Text
                style={styles.name}>{notification.eventInvite.event.name}</Text>
            <CustomButton
                text="Zaakceptuj"
                type="PRIMARY"
                onPress={() => acceptInvite(notification.eventInvite)}
                bgColor={undefined}
                fgColor={undefined}
            />
            <CustomButton
                text="Odrzuć"
                type="PRIMARY"
                onPress={() => declineInvite(notification.eventInvite)}
                bgColor={undefined}
                fgColor={undefined}
            />
        </View>
    )};

const NotificationList = ({inviteData}) => {


    return (
        <SectionList
            sections={inviteData}
            keyExtractor={(item, index) => item.id.toString()}
            renderSectionHeader={({section: {title}}) => (
                <View style={styles.sectionHeader}>
                    <Text style={styles.sectionHeaderText}>{title}</Text>
                </View>
            )}
            renderItem={({item}) => {
                if (item.notificationType === 'friendInvite') return ( <FriendInviteItem notification={item} />)
                if (item.notificationType === 'eventInvite') return ( <EventInviteItem notification={item} />)
            }}
            renderSectionFooter={({section}) =>
                section.data.length === 0 && (
                    <View style={styles.noEventsContainer}>
                        <Text style={styles.noEventsText}>Brak wydarzeń</Text>
                    </View>
                )
            }
        />
    )
};

const styles = StyleSheet.create({
    notificationContainer: {
        padding: 16,
        marginVertical: 4,
        backgroundColor: "#131417",
        borderRadius: 20,
        shadowColor: "black",
        shadowOffset: {width: 0, height: -3},
        shadowOpacity: 0.3,
        shadowRadius: 4,
    },
    name: {
        fontSize: 18,
        fontWeight: "bold",
        color: "#fff",
    },
    title: {
        fontSize: 18,
        fontWeight: "bold",
        color: "#fff",
    },
    date: {
        color: "#888",
    },
    location: {
        color: "#888",
    },
    description: {
        marginTop: 8,
    },
    sectionHeader: {
        color: "#fff",
        padding: 8,
        marginTop: 12,
    },
    sectionHeaderText: {
        fontSize: 21,
        fontWeight: "bold",
        color: "#fff",
    },
    noEventsContainer: {
        padding: 16,
        alignItems: "center",
    },
    noEventsText: {
        fontSize: 16,
        color: "#888",
    },
});

export default NotificationList;
