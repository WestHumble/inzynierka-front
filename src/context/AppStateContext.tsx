import React, {createContext, useContext, useEffect, useRef, useState} from 'react';
import axios from 'axios';
import {useNavigation} from "@react-navigation/native";
import {AppState} from "react-native";
import {EventsContext} from "./EventsContext";
import {FriendsContext} from "./FriendsContext";
export const AppContext = createContext(null);

export const AppProvider = ({children}) => {
    const {getPendingEventInvites} = useContext(EventsContext);
    const {getFriendList, getPendingFriendInvites} = useContext(FriendsContext);
    const appState = useRef(AppState.currentState);

    useEffect(() => {
        const subscription = AppState.addEventListener('change', nextAppState => {
            if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
                getPendingEventInvites()
                getFriendList()
                getPendingFriendInvites()
            }
            appState.current = nextAppState;
        });

        return () => {
            subscription.remove();
        };
    }, []);
    return (
        <AppContext.Provider value={{}}>
            {children}
        </AppContext.Provider>
    );
}