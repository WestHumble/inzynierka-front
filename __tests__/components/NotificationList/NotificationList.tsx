import React, {useState} from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from '@jest/globals';
import NotificationList from "../../../src/components/NotificationList";
import {FriendsContext} from "../../../src/context/FriendsContext";
import {ApiContext} from "../../../src/context/ApiContext";
import {EventsContext} from "../../../src/context/EventsContext";
import {NavigationContext} from "../../../src/context/NavigationContext";

test('renders empty', () => {
    const tree = renderer.create(<NotificationList inviteData={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders multiple', () => {
    const get = async (endpoint, params? = null, success = null, error = null, noAuth = false, customConfig = null)=> {}
    const post = async (endpoint, params? = null, success = null, error = null, noAuth = false, customConfig = null)=> {}

    const getPendingEventInvites = ()=>{}
    const loadAllEvents = ()=>{}
    const clearSearchEvents = ()=>{}
    const friends = [];
    const userToken = null;
    const navigationRef = React.createRef();

    const tree = renderer.create(
        <NavigationContext.Provider value={{navigationRef}}>
            <ApiContext.Provider value={{get, post, userToken}}>
                    <FriendsContext.Provider value={{friends}}>
                        <EventsContext.Provider value={{ getPendingEventInvites, loadAllEvents, clearSearchEvents}}>
                            <NotificationList inviteData={[
                                {
                                    title: "Zaproszenia na wydarzenia",
                                    data: [
                                        {
                                            id: 1,
                                            title: "title",
                                            notificationType: 'eventInvite',
                                            eventInvite: {
                                                id: 1,
                                                status: "new",
                                                event: {id:1},
                                            }
                                        }
                                    ],
                                },{
                                    title: "Zaproszenia do znajomych",
                                    data: [
                                        {
                                            id: 1,
                                            title: "title",
                                            notificationType: 'friendInvite',
                                            friendInvite: {
                                                id: 1,
                                                friend: {id:1},
                                                status: "new",
                                            }
                                        }
                                    ],
                                },
                        ]} />
                        </EventsContext.Provider>
                    </FriendsContext.Provider>
            </ApiContext.Provider>
        </NavigationContext.Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});