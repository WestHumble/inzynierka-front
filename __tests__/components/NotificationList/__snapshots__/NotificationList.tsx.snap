// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
>
  <View />
</RCTScrollView>
`;

exports[`renders multiple 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [
          {
            "eventInvite": {
              "event": {
                "id": 1,
              },
              "id": 1,
              "status": "new",
            },
            "id": 1,
            "notificationType": "eventInvite",
            "title": "title",
          },
        ],
        "title": "Zaproszenia na wydarzenia",
      },
      {
        "data": [
          {
            "friendInvite": {
              "friend": {
                "id": 1,
              },
              "id": 1,
              "status": "new",
            },
            "id": 1,
            "notificationType": "friendInvite",
            "title": "title",
          },
        ],
        "title": "Zaproszenia do znajomych",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    [
      0,
      3,
    ]
  }
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
            }
          }
        >
          Zaproszenia na wydarzenia
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "flexDirection": "row",
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "color": "#fff",
              "flex": 0.6666666666666666,
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        />
        <View
          style={
            {
              "flex": 0.3333333333333333,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              {
                "marginRight": 10,
              }
            }
          >
            <Image
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            <Image
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
            }
          }
        >
          Zaproszenia do znajomych
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "flexDirection": "row",
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 0.6666666666666666,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              onError={[Function]}
              source={
                {
                  "uri": "https://symfony-server-kg.loca.lt/api/avatar/get/1",
                }
              }
              style={
                {
                  "backgroundColor": "#F2B138",
                  "borderRadius": 5,
                  "height": 50,
                  "marginRight": 10,
                  "padding": 1,
                  "shadowColor": "black",
                  "width": 50,
                }
              }
            />
          </View>
          <Text
            style={
              {
                "color": "#fff",
                "flex": 0.6666666666666666,
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            {
              "flex": 0.3333333333333333,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              {
                "marginRight": 10,
              }
            }
          >
            <Image
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            <Image
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
